### ISSUE TITLE
[Feature request] New tool to trim the selected area if it has same color background.

### ISSUE THREAD

**Issue by mmahmoudian** [flameshot-org/flameshot#1490]
## The issue
I generally use Flameshot to copy some graphs and figures and I have to spend quite some time to move the selection area to trim the extra blank area.

## The proposed solution
Having a feature to trim the selected regions to the minimum if the background is from the same color. The following is an example I quickly made:

![image](https://user-images.githubusercontent.com/390889/111507559-7632ad00-8753-11eb-8d0e-607b16ab04af.png)

## Caveats

If there is a subtle color gradient or difference it will not be trimmed and the user might get frustrated.

**Comment by ZetaoYang** [comment]
@mmahmoudian I don't think we need to add an extra tool, it was actually implemented on the [opencv ](https://github.com/flameshot-org/flameshot/tree/opencv) branch a long time ago, and that is the **active window detection**, it will automatically trim the selected area.

**Comment by mmahmoudian** [comment]
But shouldn't active windows detection somehow query the Window Manager instead of trying to do image processing? For example in case of tiling WM such image processing would fail.

Regarding my suggestion, some hours ago I had an idea of passing the raw screenshot to imagemagic and use that to trim the white space. I'll try it today and if it work I'll probably retract my feature request.

**Comment by mmahmoudian** [comment]
I managed to put together the following:
```sh
flameshot gui -r | convert png:- -trim png:- | xclip -selection clipboard -t image/png
```

This assumes you have the following installed:
- flameshot
- imagemagic
- xclip


-------

@ZetaoYang @borgmanJeremy @thepurple 

Do you think it has merit to keep this feature request or it should be closed?

**Comment by daneel6672** [comment]
I'd definitely like this feature as I have to do this trimming manually all the time (have a gimp window open, paste to gimp, crop and export). A single shortcut should really be able to do this e.g. Snagit has it but I don't use Windows, Shutter had it but is unsupported now. That's because this is a common use case.

**Comment by edmundlaugasson** [comment]
Please allow also manually crop. Sometimes it is needed to do it manually, e.g. keep a bit more free edge and not crop too closely to content.
